sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/src/Listeners/ServiceBusListener.cs

```diff
@@ -265,122 +281,142 @@ internal void StartMessageBatchReceiver(CancellationToken cancellationToken)
-                                    foreach (ServiceBusReceivedMessage message in messagesArray)
+                                    // skip messages that were settled in the user's function
+                                    if (input.MessageActions.SettledMessages.Contains(message))
                                     {
-                                        abandonTasks.Add(receiver.AbandonMessageAsync(message, cancellationToken: cancellationToken));
+                                        continue;
                                     }
-                                    await Task.WhenAll(abandonTasks).ConfigureAwait(false);
+
+                                    // Pass CancellationToken.None to allow abandon to finish even when shutting down
+                                    abandonTasks.Add(receiver.AbandonMessageAsync(message, cancellationToken: CancellationToken.None));
                                 }
-                            }
-                        }
-                        else
-                        {
-                            // Close the session and release the session lock after draining all messages for the accepted session.
-                            if (_isSessionsEnabled)
-                            {
-                                // Use CancellationToken.None to attempt to close the receiver even when shutting down
-                                await receiver.CloseAsync(CancellationToken.None).ConfigureAwait(false);
+
+                                await Task.WhenAll(abandonTasks).ConfigureAwait(false);
                             }
                         }
                     }
-                    catch (ObjectDisposedException)
+                    else
                     {
-                        // Ignore as we are stopping the host
+                        // Close the session and release the session lock after draining all messages for the accepted session.
+                        if (_isSessionsEnabled)
+                        {
+                            // Use CancellationToken.None to attempt to close the receiver even when shutting down
+                            await receiver.CloseAsync(CancellationToken.None).ConfigureAwait(false);
+                        }
                     }
-                    catch (Exception ex)
-                    {
-                        // Log another exception
-                        _logger.LogError(ex, $"An unhandled exception occurred in the message batch receive loop");
+                }
+                catch (ObjectDisposedException)
+                {
+                    // Ignore as we are stopping the host
+                }
+                catch (TaskCanceledException)
+                    when(cancellationToken.IsCancellationRequested)
+                {
+                    // Ignore as we are stopping the host
+                    _logger.LogInformation("Message processing has been stopped or cancelled");
```

Comment:
Do we want this log?

Reply:
Yes, if we get here it means that we didn't log this message already.